demo/api/src/importer/decorators/csv-column.decorator.ts

```diff
@@ -0,0 +1,76 @@
+import { Transform } from "class-transformer";
+import { parse, parseISO } from "date-fns";
+
+import { ImporterInputClass } from "../importer-input.type";
+
+export enum Type {
+    String = "String",
+    Integer = "Integer",
+    Float = "Float",
+    Boolean = "Boolean",
+    DateTime = "DateTime",
+}
+
+export interface ParsingOptions {
+    valueMapping?: ValueMapping;
+    dateFormatString?: string;
```

Comment:
These are only available for boolean and date time, can the type reflect that?

Reply:
What do you mean?

Reply:
I can specify a value mapping even if the type is string, so the value mapping will be never used. If we would use a type like `{ type: "string" } | { type: "boolean", valueMapping ... }`, this wouldn't be possible. But I'm not sure if it's a good idea.

Reply:
We could do that or enable the use of the value mapping for any type. This could be easily done with the usual transform though

Reply:
We can leave it like this for now.